app/scripts/controllers/utils/utils.js

```diff
@@ -0,0 +1,34 @@
+///: BEGIN:ONLY_INCLUDE_IN(flask)
+import { satisfies as satisfiesSemver } from 'semver';
+import { SNAP_BLOCKLIST } from '../../constants/snaps-blocklist';
+///: END:ONLY_INCLUDE_IN
+
+/**
+ * Checks if provided snaps are on the block list.
+ *
+ * @param snapsToCheck - An object containing snap ids and other information.
+ * @returns An object structure containing snaps block information.
+ */
+async function checkSnapsBlockList(snapsToCheck) {
+  return Object.entries(snapsToCheck).reduce((acc, [snapId, snapInfo]) => {
+    const blockInfo = SNAP_BLOCKLIST.find(
+      (blocked) =>
+        (blocked.id === snapId &&
+          satisfiesSemver(snapInfo.version, blocked.versionRange, {
+            includePrerelease: true,
+          })) ||
+        (blocked.shasum ? blocked.shasum === snapInfo.shasum : false),
```

Comment:
This could probably be just `blocked.shashum === snapInfo.shasum`. If `blocked.shasum` is `undefined`, it'll just return false. And `snapInfo.shasum` is always defined.

Reply:
I had some intentions here with that approach.

If `blocked.shashum` is undefined because it's omitted from the `BLOCKLIST` object and for any reason `SnapController` returns `undefined` for `shasum` it will end up with checking `if undefined === undefined` which will be `true` and that means that any snap will be blocked. In the case where every time `shasum` is undefined (because maybe SnapController is broken because of a bad change) **every** possible snap will be **blocked**.

So, I was trying to be protective on this place just in case. Because, `extension` and `SnapsController` are two different actors communicating over messaging system where message should/can be trusted or not... If this was working together with functions and protected by types I would probably think differently.

Anyway, I can change it to simple comparison without check for `undefined`, if you think that's better, but just need to explain potential problems or consequences.

Reply:
Defensive programming is good for security parts. Seems like a good reason - could you put  a small comment explaining it in code?

Reply:
Added some comments to explain this case.